File: api_server/models/tag.py
from . import db, database


class Tag(database.Model):
    tag_id = db.Column(db.Integer, autoincrement=True, primary_key=True)
    tag_name = db.Column(db.String(64), unique=True)

    def __init__(self, name):
        self.tag_name = name.replace(' ', '-')

File: RoseliaCore/Database/Models/Tag.fs
namespace RoseliaBlog.RoseliaCore.Database.Models

open System.ComponentModel.DataAnnotations
open System.ComponentModel.DataAnnotations.Schema
open System.Text.RegularExpressions
open Microsoft.EntityFrameworkCore

[<CLIMutable>]
[<Table("tag")>]
[<Index("TagName", IsUnique = true)>]
type Tag = {
    [<Key; Column("tag_id")>]
    TagId: int
    
    [<Column("tag_name"); MaxLength(64)>]
    TagName: string
}

module Tag =
    let GetDisplayedName name =
        Regex.Replace(name, @"\s+", "-")
